WeakAurasOptions/AceGUI-Widgets/AceGUIWidget-WeakAurasDisplayButton.lua

```diff
@@ -436,57 +428,56 @@ local methods = {
             end
         end
 
-        function self.callbacks.OnDragStart(_, _)
+        function self.callbacks.OnDragStart(frame, _)
             if WeakAuras.IsImporting() or not data.parent then return end;
 
-            data.dragging = true
+            local x, y = GetCursorPosition()
+            local scale = UIParent:GetScale()
 
-            self.frame:StartMoving()
-            self.frame:SetUserPlaced(false)
+            frame.isDragged = true
+            frame:StartMoving()
+            frame:ClearAllPoints()
+            frame:SetPoint("Center", UIParent, "BOTTOMLEFT", x/scale, y/scale)
+
+            WeakAuras:SortDisplayButtons() -- find a better way to trigger a DoLayout call on the scroll frame
         end
 
-        function self.callbacks.OnDragStop(_)
-            if data.dragging then
-                self.frame:StopMovingOrSizing()
+        function self.callbacks.OnDragStop(frame)
+            if frame.isDragged then
+                frame.isDragged = false
+                frame:StopMovingOrSizing()
 
                 local id, parent = data.id, data.parent
                 local parentData = WeakAuras.GetData(parent)
                 local children = parentData.controlledChildren
 
-                local i
-                for key, value in pairs(children) do
-                    if value == id then
-                        i = key; break
-                    end
-                end
+                local i, v
+                repeat
+                    i, v = next(children, i)
+                until not i or v == id
                 if i then
                     if IsOverDisplay(parent) then
                         tremove(children, i)
                         tinsert(children, 1, id)
                         WeakAuras.Add(parentData);
-                        UpdateGroupOrder(children)
+                        WeakAuras.UpdateGroupOrders(parentData)
                     else
-                        local j
-                        for key, value in pairs(children) do
-                            if key ~= i and IsOverDisplay(value) then
-                                j = key
-                                break
-                            end
-                        end
+                        local j, v
+                        repeat
+                            j, v = next(children, j)
+                        until not j or j ~= i and IsOverDisplay(v)
                         if j then
                             tremove(children, i)
                             tinsert(children, j+1, id)
```

Comment:
I will test both ways later to see which on behaves the right way.
I must admit i only tested with a group with three children since i did not want to corrupt anything complicated.